aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -574,6 +586,24 @@ module aptos_framework::staking_contract {
             &mut store.switch_operator_events,
             SwitchOperatorEvent { pool_address, old_operator, new_operator }
         );
+
+        // Set the beneficiary address to the new operator by default.
+        set_beneficiary_for_operator_internal(new_operator, staker_address, new_operator);
+    }
+
+    public entry fun set_beneficiary_for_operator(operator: &signer, staker: address, new_beneficiary: address) acquires Store {
+        let operator_addr = signer::address_of(operator);
+        assert_staking_contract_exists(staker, operator_addr);
+        assert_account_is_registered_for_apt(new_beneficiary);
```

Comment:
Not needed if we do aptos_account::deposit_coins instead of coin::deposit as the former does registration implicitly if needed.

Reply:
Done.